File: app/services/__mocks__/appConfiguration.ts
import { IAppConfiguration, IAppConfigurationData } from 'services/appConfiguration';
import { BehaviorSubject } from 'rxjs';

const getAppConfiguration = (): IAppConfiguration => ({
  data: getAppConfigurationData()
});

export const getAppConfigurationData = (attributes = {}): IAppConfigurationData => ({
    id: 'c4b400e1-1786-5be2-af55-40730c6a843d',
    type: 'tenant',
    attributes: {
      name: 'wonderville',
      host: 'wonderville.com',
      settings: {
        core: {
          allowed: true,
          enabled: true,
          locales: ['en'],
          timezone: 'Europe/Brussels',
          organization_name: {
            en: 'Wonderville'
          },
          organization_type: 'medium_city',
          organization_site: 'https://www.wonder.ville',
          lifecycle_stage: 'active',
          color_main: '#225522',
          color_secondary: '#551122',
          color_text: '#222222',
          currency: 'EUR',
          segment_destinations_blacklist: null
        },
        participatory_budgeting: {
          allowed: true,
          enabled: true,
        }
      },
      logo: {
        small: 'http://zah.cy/wof.jpg',
        medium: 'http://fepe.et/fivacsok.jpg',
        large: 'http://jostoska.gt/timihosin.jpg',
      },
      header_bg: {
        small: 'http://neena.gr/owocap.jpg',
        medium: 'http://hedopewa.mw/ra.jpg',
        large: 'http://sarmuvov.co/uzro.jpg',
      },
      ...attributes,
    },
});

let mockAppConfiguration: IAppConfiguration = getAppConfiguration();

export const __setMockAppConfiguration = (appConfiguration: IAppConfiguration) => {
  mockAppConfiguration = appConfiguration;
};

export const currentAppConfigurationStream = jest.fn(() => {
  const observable = new BehaviorSubject(mockAppConfiguration);
  return {
    observable
  };
});

File: back/spec/factories/app_configuration.rb
require 'faker'

FactoryBot.define do
  factory :app_configuration do
    host { "localhost" }
    settings {
      {
        "core" => {
          "enabled" => true,
          "allowed" => true,
          "organization_type" => "medium_city",
          "organization_name" => {
            "en" => Faker::Address.city,
            "nl-BE" => Faker::Address.city,
            "fr-FR" => Faker::Address.city
          },
          "timezone" => "Brussels",
          "currency" => "EUR",
          "locales" => ["en","nl-BE","fr-FR"],
          "color_main" => "#335533",
          "color_secondary" => Faker::Color.hex_color,
          "color_text" => Faker::Color.hex_color,
        },
        "initiatives" => {
          "enabled" => true,
          "allowed" => true,
          "voting_threshold" => 300,
          "days_limit" => 90
        }
      }
    }

    initialize_with { AppConfiguration.send(:new, **attributes) }
  end

  factory :test_app_configuration, class: AppConfiguration do
    host { "example.org" }
    settings {
      {
          "core" => {
              "allowed" => true,
              "enabled" => true,
              "organization_type" => "medium_city",
              "organization_name" => {
                  "en" => "Liege",
                  "nl-NL" => "Luik",
                  "fr-FR" => "Liege"
              },
              "lifecycle_stage": "active",
              "locales" => ["en","fr-FR","nl-NL"],
              "timezone" => "Brussels",
              "currency" => "EUR",
              "color_main" => "#335533",
              "color_secondary" => Faker::Color.hex_color,
              "color_text" => Faker::Color.hex_color,
          },
          "initiatives" => {
              "enabled" => true,
              "allowed" => true,
              "voting_threshold" => 300,
              "days_limit" => 90
          }
      }
    }

    initialize_with { AppConfiguration.send(:new, **attributes) }
  end
end
